bundles/org.openhab.core.io.rest.voice/src/main/java/org/eclipse/smarthome/io/rest/voice/internal/VoiceResource.java

```diff
@@ -169,6 +183,91 @@ public Response interpret(@HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) @ApiParam(va
+        for (Voice voice : voices) {
+            dtos.add(VoiceMapper.map(voice));
+        }
+        return Response.ok(dtos).build();
+    }
+
+    @GET
+    @Path("/defaultvoice")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Gets the default voice.", response = VoiceDTO.class)
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 404, message = "No default voice was found.") })
+    public Response getDefaultVoice() {
+        Voice voice = voiceManager.getDefaultVoice();
+        if (voice != null) {
+            VoiceDTO dto = VoiceMapper.map(voice);
+            return Response.ok(dto).build();
+        } else {
+            return JSONResponse.createErrorResponse(Status.NOT_FOUND, "Default voice not found");
+        }
+    }
+
+    @POST
+    @Path("/say")
+    @Consumes(MediaType.TEXT_PLAIN)
+    @ApiOperation(value = "Speaks a given text with the default voice through the default audio sink.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 404, message = "No default voice or no default audio sink was found.") })
+    public Response say(@ApiParam(value = "text to speak", required = true) String text) {
+        Voice voice = voiceManager.getDefaultVoice();
+        AudioSink sink = audioManager.getSink();
+        if (voice == null) {
+            return JSONResponse.createErrorResponse(Status.NOT_FOUND, "Default voice not found");
+        } else if (sink == null) {
+            return JSONResponse.createErrorResponse(Status.NOT_FOUND, "Default audio sink not found");
+        } else {
+            voiceManager.say(text, voice.getUID(), sink.getId());
+            return Response.ok(null, MediaType.TEXT_PLAIN).build();
+        }
+    }
+
+    @POST
+    @Path("/say/{sinkid: [a-zA-Z_:0-9]*}")
+    @Consumes(MediaType.TEXT_PLAIN)
+    @ApiOperation(value = "Speaks a given text with the default voice through the given audio sink.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 404, message = "No default voice was found or audio sink does not exist.") })
+    public Response say(@ApiParam(value = "text to speak", required = true) String text,
+            @PathParam("sinkid") @ApiParam(value = "audio sink id", required = true) String sinkId) {
+        Voice voice = voiceManager.getDefaultVoice();
+        if (voice == null) {
+            return JSONResponse.createErrorResponse(Status.NOT_FOUND, "Default voice not found");
+        } else if (audioManager.getSink(sinkId) == null) {
+            return JSONResponse.createErrorResponse(Status.NOT_FOUND, "Undefined audio sink");
+        } else {
+            voiceManager.say(text, voice.getUID(), sinkId);
+            return Response.ok(null, MediaType.TEXT_PLAIN).build();
+        }
+    }
+
+    @POST
+    @Path("/say/{sinkid: [a-zA-Z_:0-9]*}/{voiceid: [a-zA-Z_:0-9]*}")
```

Comment:
Same here: Make sinkid and voiceid to QueryParams (this will actually allow you to combine the three methods into one as two params can both be optional).

Reply:
Is it possible to have QueryParam in a POST command ?

Reply:
I would think/hope so.

Reply:
I have a doubt. I will check if we have such case in all our REST APIs (POST).

Reply:
Doing a WEB search, it looks like this is possible.